lib/core/src/Cardano/Pool/Metrics.hs

```diff
@@ -110,17 +111,16 @@ data StakePool = StakePool
 --
 -- The pool productions and stake distrubtions in the db can /never/ be from
 -- different forks such that it's safe for readers to access it.
---
--- FIXME: This last statement is only true if 'putStakeProduction' and
--- 'putPoolProduction' are running in the same db transaction, which would be
--- the case if we make 'SqlPersistT' the top-level monad for the 'DBLayer'
--- instead of 'IO'.
 monitorStakePools
     :: Trace IO Text
     -> NetworkLayer IO t Block
     -> DBLayer IO
     -> IO ()
-monitorStakePools tr nl db = do
+monitorStakePools
+    tr
+    nl
+    DBLayer{atomically,readPoolProductionCursor,rollbackTo,
+    putStakeDistribution, putPoolProduction} = do
```

Comment:
Moving this pattern-matching into a where clause would make the overall more readable and keep the top-level signature and function head clean and neat.